### ISSUE TITLE
Backoffice extension source files do not see inherited methods

### ISSUE THREAD

**Issue by agrabovskis** [epam/sap-commerce-intellij-idea-plugin#729]
The specific case was reproduced with classes being in different extensions:
* backoffice - BulkEditForm
* platformbackoffice - ClassificationBulkEditForm (extends BulkEditForm)

Consider the following code example:
```
		BulkEditForm backofficeForm = new BulkEditForm();
		backofficeForm.getTemplateObject(); // BulkEditForm method - OK

		// ClassificationBulkEditForm extends BulkEditForm
		ClassificationBulkEditForm platformbackofficeForm = new ClassificationBulkEditForm();
		platformbackofficeForm.getClassificationAttributesForm(); // ClassificationBulkEditForm method - OK
		platformbackofficeForm.getTemplateObject(); // Cannot resolve method 'getTemplateObject' in 'ClassificationBulkEditForm'

		backofficeForm = platformbackofficeForm; // Incompatible types. Found: 'de.hybris.platform.platformbackoffice.bulkedit.ClassificationBulkEditForm', required: 'com.hybris.backoffice.bulkedit.BulkEditForm'
```

* Code completion indicates errors as given above
* The code compiles (!)
* The classes are navigatable (Cmd + Click on ClassificationBulkEditForm navigates correctly to target location and Cmd + Click on BulkEditForm work just fine).

Attaching an extension (after `ant clean`) showcasing the issue.
[demobackoffice.zip](https://github.com/epam/sap-commerce-intellij-idea-plugin/files/12645831/demobackoffice.zip)

I tried setting up the module dependencies manually, but didn't manage to make the above code fragment to compile in IntelliJ (it works just fine with `ant all`).

While checking the dependencies I got an impression that the backoffice module dependencies in the new system might require revising. The new system of sub-modules is improvement over the previous one as allows a more fine-grained control of exports for dependency management, but here are the things that I find wrong (please let me know if I've missed any caveats):
1. Adding dependency to `Custom.demo.demobackoffice -> Hybris.backoffice-framework.backoffice.web` and exporting it "leaks" backoffice classes to core extensions. The correct should be `Custom.demo.demobackoffice.backoffice -> Hybris.backoffice-framework.backoffice.web` (if there is a dependency to `backoffice` extension).
2. `Hybris.backoffice-framework.backoffice.web` should not be added in every case of `backoffice-module = true` but only when there is also a dependency `demobackoffice -> backoffice`. In the demo extension I attached to this case there is a dependency path `demobackoffice -> platformbackoffice -> backoffice` - dependencies should be transitive, hence `platformbackoffice.backoffice -> backoffice.web` and `demobackoffice.backoffice -> platformbackoffice.backoffice` should do the trick.

**Comment by mlytvyn** [comment]
@agrabovskis , wow, so nice and detailed report, just incredible! thank you for highlighting this.

there are few improvements related to compilation in `2023.2.9`, will try your case on it and return to you.

**Comment by mlytvyn** [comment]
@agrabovskis , indeed, I can reproduce it and confirm this case.

I'm thinking of adding `backoffice.web - Web Classes` also to the global `Backoffice Library`.

Please, wait for release `2023.2.9`, I'll try my best to adjust class path accordingly.

**Comment by agrabovskis** [comment]
@mlytvyn, many thanks and happy to help with detailed descriptions if nothing else :)

Another option would be to set up module dependencies based on extension info - that way the autocomplete would suggest only the classes from dependent extensions (see image below). But it might be a bit tricky to get it right (or even unrealistic), hence might be left for later or rejected for good.

![attels](https://github.com/epam/sap-commerce-intellij-idea-plugin/assets/842013/a9516442-062f-445e-a77a-7b0458f8b38f)

**Comment by mlytvyn** [comment]
@agrabovskis , looks like IntelliJ IDEA cannot have both classes and jars set for the same project Library.
When I pack backoffice.web web classes into the jar file and add it to the Backoffice Library - all classes become visible.
But when I attach those classes as a folder - nope.

dirty hack -> pack those classes into jar file and use in the Backoffice Library

**Comment by agrabovskis** [comment]
Sounds more like a bug of IntelliJ. If reproducible, then maybe it's wort filing a bug?

**Comment by mlytvyn** [comment]
@agrabovskis , indeed, looks so. I'll create new ticket in [youtrack](https://youtrack.jetbrains.com/issues).
p.s. according to docs it should work with both classes in folder and jars https://www.jetbrains.com/help/idea/library.html#lib_levels

**Comment by mlytvyn** [comment]
@agrabovskis , I was able to reproduce it for simplest possible java project.
Attached that sample project in the newly reported issue to JetBrains https://youtrack.jetbrains.com/issue/IDEA-332845/Project-Library-ignores-classes-in-the-folders.

Besides that...

- I'll adjust the Plugin code in a way `BackOffice Library` will also use classes from the `backoffice.web - classes` directory.
- Then, we have to wait for JetBrains fix, I'll also contact them in JetBrains' Plugin development Slack.
- If fix will be released... some time in xxx years, we have to pack those classes into a jar as I wrote above, yeah, I know, not ideal, but nothing else comes to my tired mind right now.

**Comment by mlytvyn** [comment]
@agrabovskis , let's wait for JetBrains on this. Unfortunately I do not have time to fix it only in Plugin, all solutions look like hacks, which will be hard to detect/remove/refactor.

thank you once again for reporting this issue, I applied at least few other changes which may improve our day-to-day life.

**Comment by agrabovskis** [comment]
Totally agree! No need to implement any workarounds that might be hard to maintain in the future. Let's wait on IntelliJ team now.

Thank you very much for addressing the issue.
